esmvaltool/diag_scripts/shared/_validation.py

```diff
@@ -63,9 +66,15 @@ def get_control_exper_obs(short_name, input_data, cfg, cmip_type):
     input_data: dict containing the input data info
     cfg: config file as used in this module
     """
-    # select data per short name and CMIP type
-    dataset_selection = select_metadata(
-        input_data, short_name=short_name, project=cmip_type)
+    # nocmip
+    # # select data per short name and CMIP type
+    # dataset_selection = select_metadata(
+    #     input_data, short_name=short_name, project=cmip_type)
+
+    # select data per short name and optional CMIP type
+    dataset_selection = (select_metadata(input_data, short_name=short_name,
+                                         project=cmip_type) if cmip_type else
+                         select_metadata(input_data, short_name=short_name))
```

Comment:
could you maybe expand via a conditional on `cmip_type` eg `if not cmip_type: ... else: ...` - code reads better that way

Reply:
Hi @valeriupredoi, Thank you very much for reviewing this. I looked into your comments. I deleted much of the commented code. However I am actually unsure for the function _disentagle_iden_datasets, whether this is used in a different diagnostic?

Reply:
I changed the parts you suggested. I hope I understood everything correctly... I am not a full-blood coder. Tell me if there is something I missed.

Reply:
excellent, very many thanks! No need to be a full-blood coder, although I do like the expression :laughing: Did you forget to push to remote, though? I don't see any more changes after my last commit last week

Reply:
I made some additional checks and pushed it today.

Reply:
great, cheers! I'll have a look now and test run the validation recipe as it is in bog standard in the recipes :beer: